[9.0][wasm] Unable to find package Microsoft.NETCore.App.Ref with version (= 8.0.11) or (= 6.0.36)

## Build Information

Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=861525
Build error leg or test failing: NoWorkload-NoWebcil-ST-Wasm.Build.Tests

## Error Message

```json
{
  "ErrorMessage" : "error NU1102: Unable to find package Microsoft.NETCore.App.Ref with version",
  "BuildRetry" : false,
  "ExcludeConsoleLog" : false
}
```

- PR: https://github.com/dotnet/runtime/pull/109563
- Log file: https://helixre107v0xd1eu3ibi6ka.blob.core.windows.net/dotnet-runtime-refs-pull-109563-merge-3b35c9781aba4eed96/NoWorkload-NoWebcil-ST-Wasm.Build.Tests/1/console.a41298c9.log?helixlogtype=result
- Output:
```
    Wasm.Build.Tests.NonWasmTemplateBuildTests.NonWasmConsoleBuild_WithoutWorkload(config: "Release", extraBuildArgs: "/p:WasmBuildNative=true", targetFramework: "net8.0") [FAIL]
       Expected 0 exit code but got 1: /root/helix/work/workitem/e/dotnet-none/dotnet build -restore -c Release -bl:/root/helix/work/workitem/uploads/xharness-output/logs/nonwasm_net8.0_Release_cgbvwf5i_dog.binlog /p:WasmBuildNative=true -f net8.0
      Standard Output:
      []   Determining projects to restore...
      []   Restored /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj (in 290 ms).
      []   Determining projects to restore...
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Ref with version (= 8.0.11)
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1458 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1417 version(s) in dotnet9 [ Nearest version: 9.0.0-alpha.1.23423.2 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 127 version(s) in dotnet-public [ Nearest version: 9.0.0-preview.1.24080.9 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1 version(s) in nuget-local [ Nearest version: 9.0.1 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Ref with version (= 8.0.11)
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1593 version(s) in dotnet8 [ Nearest version: 9.0.0-alpha.1.23414.15 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1149 version(s) in dotnet9 [ Nearest version: 9.0.0-alpha.1.23417.4 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 128 version(s) in dotnet-public [ Nearest version: 9.0.0-preview.1.24081.5 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 0 version(s) in nuget-local
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Host.linux-x64 with version (= 8.0.11)
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1458 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1417 version(s) in dotnet9 [ Nearest version: 9.0.0-alpha.1.23423.2 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 176 version(s) in dotnet-public [ Nearest version: 9.0.0-preview.1.24080.9 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 0 version(s) in nuget-local
      []   Failed to restore /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj (in 762 ms).
      [] 
      [] Build FAILED.
      [] 
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Ref with version (= 8.0.11)
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1458 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1417 version(s) in dotnet9 [ Nearest version: 9.0.0-alpha.1.23423.2 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 127 version(s) in dotnet-public [ Nearest version: 9.0.0-preview.1.24080.9 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1 version(s) in nuget-local [ Nearest version: 9.0.1 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Ref with version (= 8.0.11)
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1593 version(s) in dotnet8 [ Nearest version: 9.0.0-alpha.1.23414.15 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1149 version(s) in dotnet9 [ Nearest version: 9.0.0-alpha.1.23417.4 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 128 version(s) in dotnet-public [ Nearest version: 9.0.0-preview.1.24081.5 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 0 version(s) in nuget-local
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Host.linux-x64 with version (= 8.0.11)
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1458 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 1417 version(s) in dotnet9 [ Nearest version: 9.0.0-alpha.1.23423.2 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 176 version(s) in dotnet-public [ Nearest version: 9.0.0-preview.1.24080.9 ]
      [] /root/helix/work/workitem/e/wbt artifacts/nonwasm_net8.0_Release_cgbvwf5i_dog/nonwasm_net8.0_Release_cgbvwf5i_dog.csproj : error NU1102:   - Found 0 version(s) in nuget-local
      []     0 Warning(s)
      []     3 Error(s)
      [] 
      [] Time Elapsed 00:00:01.62
      
      Stack Trace:
        /_/src/mono/wasm/Wasm.Build.Tests/Common/CommandResult.cs(51,0): at Wasm.Build.Tests.CommandResult.EnsureExitCode(Int32 expectedExitCode, String messagePrefix, Boolean suppressOutput)
        /_/src/mono/wasm/Wasm.Build.Tests/Common/CommandResult.cs(28,0): at Wasm.Build.Tests.CommandResult.EnsureSuccessful(String messagePrefix, Boolean suppressOutput)
        /_/src/mono/wasm/Wasm.Build.Tests/NonWasmTemplateBuildTests.cs(116,0): at Wasm.Build.Tests.NonWasmTemplateBuildTests.NonWasmConsoleBuild(String config, String extraBuildArgs, String targetFramework, String directoryBuildTargets, Boolean shouldRun)
        /_/src/mono/wasm/Wasm.Build.Tests/NonWasmTemplateBuildTests.cs(79,0): at Wasm.Build.Tests.NonWasmTemplateBuildTests.NonWasmConsoleBuild_WithoutWorkload(String config, String extraBuildArgs, String targetFramework)
           at InvokeStub_NonWasmTemplateBuildTests.NonWasmConsoleBuild_WithoutWorkload(Object, Span`1)
           at System.Reflection.MethodBaseInvoker.InvokeWithFewArgs(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```
<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=861525
**Error message validated:** `[error NU1102: Unable to find package Microsoft.NETCore.App.Ref with version`]
**Result validation:** :x: Known issue did not match with the provided build.
**Validation performed at:** 11/7/2024 11:16:25 PM UTC
<!-- Known issue validation end -->
<!--Known issue error report start -->

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[867435](https://dev.azure.com/dnceng-public/public/_build/results?buildId=867435)|dotnet/runtime|[NoWorkload-NoWebcil-ST-Wasm.Build.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=867435&view=ms.vss-test-web.build-test-results-tab&runId=22584068&resultId=100010)|dotnet/runtime#109723|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|1|
<!--Known issue error report end -->

--- a/src/mono/wasm/Wasm.Build.Tests/NonWasmTemplateBuildTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/NonWasmTemplateBuildTests.cs
@@ -67,9 +67,13 @@ public class NonWasmTemplateBuildTests : TestMainJsTestBase
         )
         .MultiplyWithSingleArgs
         (
-            "net6.0",
-            s_previousTargetFramework,
-            s_latestTargetFramework
+            EnvironmentVariables.WorkloadsTestPreviousVersions
+                ? [
+                    "net6.0",
+                    s_previousTargetFramework,
+                    s_latestTargetFramework
+                ]
+                : [s_latestTargetFramework]
         )
         .UnwrapItemsAsArrays().ToList();
 

